### PR TITLE
chore: record artifact state & render state on resource details. Fixes #5789

### LINKED ISSUE #5789: [v2compat] log MLMD artifact state during execution

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Fixes https://github.com/kubeflow/pipelines/issues/5789
image

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: frontend/src/components/ResourceInfo.tsx [hunk 1]
```diff
@@ -71,7 +71,16 @@ export class ResourceInfo extends React.Component<ResourceInfoProps, {}> {
...
+        <h1 className={commonCss.header}>
+          {this.props.typeName}
+          {(() => {
+            const stateText = getResourceStateText(this.props);
+            if (stateText) {
+              return ` (${stateText})`;
+            }
+            return '';
+          })()}
+        </h1>
```
Comment (Bobgy) [inline frontend/src/components/ResourceInfo.tsx:83]: because we didn't log state before, if we found state=Unknown, it's better not to render it

Comment (Bobgy) [reply frontend/src/components/ResourceInfo.tsx:83]: The text change from `Type: <type-name>` to `<type-name> (<state>)` is following principle that, when text content can tell what it is, UI would feel more concise when the `Type:` and `State:` prefix is omitted.

For example, when we see `kfp.ContainerExecution`, we know it is a type of execution, so it's unnecessary to add prefix `Type: ` now.

File: frontend/src/components/ResourceInfo.tsx [hunk 2]
```diff
@@ -71,7 +71,16 @@ export class ResourceInfo extends React.Component<ResourceInfoProps, {}> {
...
-        <h1 className={commonCss.header}>Type: {this.props.typeName}</h1>
+        <h1 className={commonCss.header}>
+          {this.props.typeName}
+          {(() => {
+            const stateText = getResourceStateText(this.props);
+            if (stateText) {
+              return ` (${stateText})`;
+            }
+            return '';
+          })()}
```
Comment (zijianjoy) [inline frontend/src/components/ResourceInfo.tsx:82]: Can this part be extracted to a const var? Hope that it makes the element rendering logic more concise:

```
let resourceTitle = this.props.typeName;
const stateText = getResourceStateText(this.props);
if (stateText) {
  resourceTitle = resourceTitle + ` (${stateText})`;
}

return (
      ...
        <h1 className={commonCss.header}>
          {resourceTitle}
        </h1>
      ....
   )
```

**Review by zijianjoy — COMMENTED** [review]
/lgtm

Thank you Yuan for exposing the state info!

File: frontend/src/components/ResourceInfo.tsx (hunk 2 above)
Comment (Bobgy) [reply frontend/src/components/ResourceInfo.tsx:82]: Good point, will change as you suggested

**Comment by Bobgy** [comment]
Got error:
> F0709 05:29:17.078200       1 main.go:56] Failed to execute component: failed to download input artifact "input_dir" from remote storage URI "gs://kfp-ci/4ca1dd2d702e106386af42120c73df46ed8c0cc5/v2-sample-test/data/samples_config-loop-item/dir-pipeline-v2/dir-pipeline-v2-hd2tw/produce-dir-with-files-v2-python-op/output_dir": failed to list objects in remote storage "dir-pipeline-v2/dir-pipeline-v2-hd2tw/produce-dir-with-files-v2-python-op/output_dir": blob (code=Unknown): Get "

in output a directory sample
https://4e18c21c9d33d20f-dot-datalab-vm-staging.googleusercontent.com/#/runs/details/2eb0df4d-74c8-4a01-b466-23718e38ed23

**Pull request merged**